src/components/Footer.tsx

```diff
@@ -0,0 +1,93 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { SortType } from '../types/SortType';
+import { url } from './Urls';
+
+export const Footer: React.FC<{
+  askTodos: (url: string) => void
+  deleteCompleted: () => void
+  todosFromServer: Todo[] | undefined
+  countComplited: boolean | undefined;
+}> = ({
+  askTodos,
+  deleteCompleted,
+  todosFromServer,
+  countComplited,
+}) => {
+  const [selectedForm, setSelectedForm] = useState(SortType.All);
+
+  const sortTodos = (format: SortType) => {
+    switch (format) {
+      case SortType.Active:
+        askTodos(`${url}&completed=false`);
+        setSelectedForm(SortType.Active);
+        break;
+      case SortType.Completed:
+        askTodos(`${url}&completed=true`);
+        setSelectedForm(SortType.Completed);
+        break;
+
+      case SortType.All:
+      default:
+        askTodos(url);
+        setSelectedForm(SortType.All);
+        break;
+    }
+  };
+
+  return (
+    <>
+      <span className="todo-count">
+        {`${todosFromServer?.length || 0} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link', {
+              selected: selectedForm === SortType.All,
+            },
+          )}
+          onClick={() => sortTodos(SortType.All)}
+        >
```

Comment:
Create separate handler 
```
const onFilterChange = (filter: SortType) => () => {
  sortTodos(SortType.All)
} 
```
usage:
```suggestion
          onClick={onFilterChange(SortType.All)}
        >
```